File: src/main/java/ma/hariti/asmaa/wrm/simulator/dto/request/QuestionDTO.java
package ma.hariti.asmaa.wrm.simulator.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class QuestionDTO {
    private Long id;
    private String content;
    private String expectedAnswer;
    private Long sessionId;
    private String skill;
    private AnswerDTO answer;
}

File: frontend/simulator/src/app/models/questiondto.ts
export interface QuestionDTO {
  id: number;
  content: string;
  expectedAnswer?: string;
  sessionId: number;
  answer?: {
    content?: string;
    score?: number;
    feedback?: string;
    improvementSuggestions?: string[];
    followUpQuestion?: string;
  } | null;
}
